### PR TITLE
redis: optional prometheus metrics exporter

### PR DESCRIPTION
**Pull request by lwolf**
This PR adds the option to run a side-car Prometheus metrics exporter for redis. It requires no additional configuration and can be enabled be setting metrics.enabled=true

### CODE REVIEW

**Comment by seanknox** [comment]
@k8s-bot ok to test

File: stable/redis/templates/svc.yaml [hunk 1]
```diff
@@ -7,6 +7,11 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
+{{- if .Values.metrics.enabled }}
```
Comment (unguiculus) [inline stable/redis/templates/svc.yaml:10]: I'm not sure if we should always add these annotations when metrics are enabled. Maybe this should be configurable. You don't need these annotations if you use the Prometheus Operator.

@mgoodness What do you think?

File: stable/redis/values.yaml [hunk 2]
```diff
@@ -32,6 +32,16 @@ persistence:
   accessMode: ReadWriteOnce
   size: 8Gi
 
+metrics:
+  enabled: false
+  image: oliver006/redis_exporter
+  imageTag: v0.11
+  imagePullPolicy: IfNotPresent
+  resources:
+    requests:
```
Comment (unguiculus) [inline stable/redis/values.yaml]: Don't specify defaults for resources.

File: stable/redis/Chart.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: redis
-version: 0.5.1
+version: 0.5.2
```
Comment (unguiculus) [inline stable/redis/Chart.yaml]: I would make this a `0.6.0`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/redis/templates/svc.yaml (hunk 1 above)
Comment (mgoodness) [reply stable/redis/templates/svc.yaml:10]: I'm very much inclined to agree. How about something like:
```
{{- if .Values.metrics.enabled }}
  annotations:
{{ toYaml .Values.metrics.annotations | indent X }}
{{- end }}
```

File: stable/redis/templates/deployment.yaml [hunk 4]
```diff
@@ -45,6 +45,24 @@ spec:
         volumeMounts:
         - name: redis-data
           mountPath: /bitnami/redis
+{{- if .Values.metrics.enabled }}
+      - name: metrics
+        image: "{{ .Values.metrics.image }}:{{ .Values.metrics.imageTag }}"
+        imagePullPolicy: {{ default "" .Values.metrics.imagePullPolicy | quote }}
```
Comment (mgoodness) [inline stable/redis/templates/deployment.yaml]: Avoid using the `default` function if you can - it makes it harder for end-users to determine what the value will be. In this case, set `metrics.imagePullPolicy` explicitly in `values.yaml`.

**Review by mgoodness — CHANGES_REQUESTED** [review]

**Comment by lwolf** [comment]
updated pull request with all changes.

**Review by unguiculus — APPROVED** [review]

**Review by mgoodness — APPROVED** [review]

**Comment by mgoodness** [comment]
@unguiculus

**Pull request merged**